Library/Homebrew/cmd/install.rb

```diff
@@ -281,6 +281,33 @@ def run
           puts "You're on your own. Failures are expected so don't create any issues, please!"
         end
 
+        overwrite = if !args.overwrite? &&
+                       GitHub::Actions.env_set? &&
+                       ENV["HOMEBREW_GITHUB_ACTIONS_NO_INSTALL_OVERWRITE"].blank? &&
+                       ENV["HOMEBREW_GITHUB_ACTIONS"].blank?
+          if ENV["HOMEBREW_GITHUB_ACTIONS_NO_INSTALL_OVERWRITE_WARNING"].blank?
+            message = <<~WARNING
+              In GitHub Actions, `brew install` behaves the same as `brew install --overwrite`.
```

Comment:
Yeh, sorry to be a pain here: this definitely feels like an overly broad change. Would like to see this scoped to only Python for now and we can consider widening it in future if desired.